### ISSUE TITLE
Different upstreams for allow listed domains

### ISSUE THREAD

**Issue by timkgh** [AdguardTeam/AdGuardHome#3777]
If the upstreams themselves block some domains, allow listing them in AGH does not help, the upstreams will still return `0.0.0.0`. E.g. using the AdGuard DNS upstreams (or other filtering upstreams) for the family protection features.

It would be useful to have an optional config in AGH that uses a different upstream for allow listed items so they never get blocked.

**Comment by timkgh** [comment]
@ameshkov wdyt?

**Comment by ameshkov** [comment]
Like automatically? I actually thought that using https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-for-domains already covers this usecase.

**Comment by timkgh** [comment]
Yes, automatically, so I don't have to add both allow rules and upstreams for allowed domains. Just a global (optional) setting, for any allowed domains, use these other (unfiltered) upstream DNS servers because I really want them resolved. E.g. it is not enough to allow list some domains if I use the upstream AG DNS servers that still filter them.

**Comment by ameshkov** [comment]
Technically, this is of course possible, but seems rather specific.
Let's see if this feature request would be popular.

The alternative solution would be to further extend the filtering rules syntax and introduce something like an `$upstream` modifier.
